src/diffpy/fourigui/fourigui.py

```diff
@@ -394,8 +394,9 @@ def fft(self):
         def perform_fft(fftholder):
             fftholder = np.nan_to_num(fftholder)
             size = list(fftholder.shape)
+            axes = list(range(len(size)))
```

Comment:
this is not very clear.  In the comments, let's have a brief discussion of what is deprecated in numpy2 and how we are fixing it.  Not too much, but presumably @Fil158 you googled something and found something out so it is helpful to share what you found out and a few words about how it affects us here and so your suggested fix.  It builds more confidence that we understand what we are doing.

Also, please make your PR title more informative. Which warning does this PR fix?  Not too long, but just enough.

I am not sure but there is almost certainly a more readable fix than `list(range(len(size)))`

Reply:
Here is the worning, please see the [Numpy official documentation](https://numpy.org/doc/2.1/reference/generated/numpy.fft.fftn.html#numpy.fft.fftn)

![image](https://github.com/user-attachments/assets/21294e1c-5254-4d9d-abb5-30754ddcd5ae)

To solve it I specify axes which is a list of integers with the same size of s
![image](https://github.com/user-attachments/assets/a46fca26-27e5-4324-8712-91b5c742514c)

After the changes pytests run correctly and fourigui app works with test data.

Reply:
Thanks @Fil158 this is very helpful.  Next we need a quick discussion about what the shape of the arrays are and what th axes are and what our desired fft is.

Then finally, what code we think is best to accomplish that.